tools/lint-docstring

```diff
@@ -0,0 +1,150 @@
+#!/usr/bin/env python3
+
+import argparse
+import glob
+import importlib
+from pathlib import Path
+from typing import Dict, List, Tuple
+
+
+TABLE_OFFSET = 7
+COLUMN_WIDTHS = (23, 20, 103)
+SPACEBAR = " "
+
+# leaving blank rows after exhausting files from each folder
+BLANK_LINE = f"| {COLUMN_WIDTHS[0]*SPACEBAR}| {COLUMN_WIDTHS[1]*SPACEBAR}| {COLUMN_WIDTHS[2]*SPACEBAR}|\n"
+
+# absolute path to zulip-terminal
+ROOT_DIRECTORY = Path(__file__).resolve().parent.parent
+
+# absolute path to zulip-terminal/zulipterminal to be passed as parameter
+ZULIPTERMINAL = ROOT_DIRECTORY / "zulipterminal"
+
+# new doc file has been created for the time being to compare it with the original
+DEVELOPER_DOC = ROOT_DIRECTORY / "docs" / "developer-file-overview.md"
+
+
+# zulipterminal/scripts and zulipterminal/themes do not have any files
+# mentioned in the file-overview. Hence, their description has been
+# added separately.
+DOCSTRINGS_FOR_NO_FILE_FOLDERS = {
+    "zulipterminal/themes": "Themes bundled with the application",
+    "zulipterminal/scripts": "Scripts bundled with the application",
+}
+
+
+def main() -> None:
+    create_file_overview_doc()
+    print(f"Generated {Path(DEVELOPER_DOC).name} successfully.")
+
+
+def create_file_overview_doc() -> None:
+    """
+    Creates the entire developer-file-overview.md document again by retaining the
+    lines 1 to TABLE_OFFSET and the rest is written using the docstrings in the files
+    """
+    folder_file_docstring = generate_folder_file_docstrings_dict()
+    with open(DEVELOPER_DOC, "r") as developer_file_read:
+        doc_data = developer_file_read.readlines()
+
+    table_markdown = []
+    for folder in folder_file_docstring:
+        dictionary_of_files = folder_file_docstring[folder]
+
+        folder_text = folder
+        for file in sorted(dictionary_of_files):
+            docstring = folder_file_docstring[folder][file]
+            text_tuple = (folder_text, file, docstring)
+            padded_folder_cell, padded_file_cell, padded_desc_cell = pad_column_cells(
+                text_tuple, COLUMN_WIDTHS
+            )
+            new_line = (
```

Comment:
Do you want to use `line` here or `col(umn)`? Both are technically correct but cell, column and table fit a pattern.

Reply:
Considering the pattern, maybe call it `new_row`? As we are adding new _rows_ in each iteration?